packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -224,11 +286,19 @@ export async function sideloadAddIn(manifestPath: string, app?: OfficeApp, canPr
     }
   }
 
-  await registerAddIn(manifestPath);
-
-  if (app) {
-    const sideloadFile = await generateSideloadFile(app, manifest);
-
-    await open(sideloadFile, { wait: false });
+  if (isSideloadingSupportedForHost(app, isDesktop)) {
+    if (isDesktop) {
+      await registerAddIn(manifestPath);
+      sideloadFile = await generateSideloadFile(app, manifest, document);
+      await open(sideloadFile, { wait: false });
+    } else {
+      if (document === undefined) {
+        throw new Error("Missing --document paramater.");
+      }    
+      sideloadFile = await generateSideloadUrl(manifestPath, document, isTest);
+      await open(sideloadFile, { wait: false });
```

Comment:
The open() function is used for both desktop and web. You could move it down below the if statement,